doc/computing/randomness.rst

```diff
@@ -0,0 +1,315 @@
+    >>> rng = np.random.RandomState(0)
+    >>> cv = KFold(n_splits=2, shuffle=True, random_state=rng)
+    >>> for train, test in cv.split(X):
+    ...     print(train, test)
+    [0 3 5 6 7] [1 2 4 8 9]
+    [1 2 4 8 9] [0 3 5 6 7]
+    >>> for train, test in cv.split(X):
+    ...     print(train, test)
+    [0 4 6 7 8] [1 2 3 5 9]
+    [1 2 3 5 9] [0 4 6 7 8]
+
+We can see that the splits are different from the second time `split` is
+called. This may lead to unexpected results if you compare the performance of
+multiple estimators by calling `split` many times: the estimators will not
+be evaluated on the same folds, and the scores on each fold will not be
+comparable. On average, if enough folds are used and with enough data, one
+can however expect that that mean score allows to conclude whether one
+estimator is better than another. In any case, passing an int makes score
+comparisons across folds meaningful, and is usually safer.
+
+
+Common pitfalls and subtleties
+------------------------------
+
+While the rules that govern the `random_state` parameter are seemingly simple,
+they do however have some subtle implications. In some cases, this can even
+lead to wrong conclusions.
+
+Estimators
+..........
+
+**Differences in cross-validation procedures**
+
+Depending on what is passed as the `random_state` parameter, estimators may
+behave very differently, especially in cross-validation procedures. Consider
+the following snippet::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import cross_val_score
+    >>> import numpy as np
+
+    >>> X, y = make_classification(random_state=0)
+
+    >>> rf_inst = RandomForestClassifier(random_state=np.random.RandomState(0))
+    >>> cross_val_score(rf_inst, X, y)
+    array([0.9 , 0.95, 0.95, 0.9 , 0.9 ])
+
+    >>> rf_123 = RandomForestClassifier(random_state=123)
+    >>> cross_val_score(rf_123, X, y)
+    array([0.85, 0.95, 0.95, 0.9 , 0.9 ])
+
+We see that the cross-validated scores of `rf_inst` and `rf_123` are
+different, as should be expected since we didn't pass the same `random_state`
+parameter. However, the difference between these scores is more subtle that
+it may look, and **the cross-validation procedures that were performed by**
+:func:`~sklearn.model_selection.cross_val_score` **significantly differ in
+each case**:
+
+- Since `rf_123` was passed an int, every call to `fit` uses the same RNG:
+  the same (random) subset of features will be used across all folds to fit
+  the random forest.
+- Since `rf_inst` was passed a `RandomState` instance, each call to `fit`
+  starts from a different RNG, and the randomly sampled subset of features
+  will be different for each of the 5 folds of the CV procedure.
+
+Here, neither procedure is inherently wrong, and one might prefer one over
+the other depending on the task at hand. It is however important to
```

Comment:
Here, I would loop back to the comment that I made above:

1) to evaluate a training procedure, seeds shouldn't be fixed (because what is a good seed in a given situation may be a bad one in another situation)

2) in a train-test split, to evaluate a trained model, it can be useful to fix the seed.